docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,361 @@
+# Formatted ORAS CLI output
+
+## Table of Contents
+
+- [Background](#background)
+- [Guidance](#guidance)
+- [Scenarios](#scenarios)
+  - [Scripting](#scripting)
+  - [CI/CD](#cicd)
+- [Proposal and desired user experience](#proposal-and-desired-user-experience)
+  - [oras manifest fetch](#oras-pull)
+  - [oras pull](#oras-pull)
+  - [oras push](#oras-push)
+  - [oras attach](#oras-attach)
+  - [oras discover](#oras-discover)
+- [FAQ](#faq)
+
+## Background
+
+ORAS has prettified output designed for humans. However, for machine processing, especially in automation scenarios, like scripting and CI/CD pipelines, developers want to perform batch operations and chain different commands with ORAS, as well as filtering, modifying, and sorting objects based on the outputs that are emitted by the ORAS command. Developers expect that ORAS output can be emitted as machine-readable text instead of only the prettified or tabular data, so that it can be used to perform other operations.
+
+The formatted output is not intended to supersede the prettified human-readable and friendly output text of ORAS CLI. It aims to provide multiple views and a programming-friendly experience for developers, DevOps engineers, IT professionals who want to automate their workflows without needing to parse unstructured text.
+
+## Guidance
+
+ORAS allows to use `--output` to output a file or directory in the filesystem. To enable users to format the metadata output of ORAS commands and compute figures based on the formatted output, these two options are proposed as follows:
+
+- Use `--format <DATA_FORMAT>` to format metadata output of ORAS commands into different formats including prettified JSON, tree, table view, and Go template, i.e. `--format json|tree|table|go-template=GO_TEMPLATE`. It supports computing figures within the template using [Sprig](http://masterminds.github.io/sprig/) functions. This is the primary and recommended usage.
```

Comment:
Suggest to skip <DATA_FORMAT> since it's explained later.
```suggestion
- Use `--format` to format metadata output of ORAS commands into different formats including prettified JSON, tree, table view, and Go template, i.e. `--format json|tree|table|go-template=GO_TEMPLATE`. It supports computing figures within the template using [Sprig](http://masterminds.github.io/sprig/) functions. This is the primary and recommended usage.
```